src/hooks/loadables/useLoadBalances.ts

```diff
@@ -12,8 +12,8 @@ export const useLoadBalances = (): AsyncResult<SafeBalanceResponse> => {
   const currency = useAppSelector(selectCurrency)
 
   // Re-fetch assets when the entire SafeInfo updates
-  const [data, error, loading] = useAsync<SafeBalanceResponse>(
-    () => {
+  const [data, error, loading] = useAsync<SafeBalanceResponse | undefined>(
```

Comment:
Restored this because otherwise it keeps the old balances when navigating between safes. The same with tx history and queue (see below).